synthesizer/process/src/stack/authorization/mod.rs

```diff
@@ -62,8 +62,13 @@ impl<N: Network> TryFrom<(Vec<Request<N>>, Vec<Transition<N>>)> for Authorizatio
             requests.len(),
             transitions.len()
         );
+        // Move the first request to the back in order to match the transitions.
+        let mut requests_deque = VecDeque::from(requests.clone());
+        if let Some(first_request) = requests_deque.pop_front() {
+            requests_deque.push_back(first_request);
+        }
         // Ensure the requests and transitions are in order.
-        for (index, (request, transition)) in requests.iter().zip_eq(&transitions).enumerate() {
+        for (index, (request, transition)) in requests_deque.iter().zip_eq(&transitions).enumerate() {
```

Comment:
I'm not yet familiar with the full context behind this change (it's rather odd these are not ordered the same way by default) but just looking at the rust, you could probably do something like this to avoid building a second `VecDeque` and calling `clone`, `pop` and `insert`:
```suggestion
        for (index, (request, transition)) in requests.iter().skip(1).chain(requests.iter().take(1)).zip_eq(&transitions).enumerate() {
```